src/core/trie.js

```diff
@@ -0,0 +1,79 @@
+// @flow
+
+import {type AddressModule} from "./address";
+import {
+  NodeAddress,
+  type NodeAddressT,
+  EdgeAddress,
+  type EdgeAddressT,
+} from "./graph";
+import * as NullUtil from "../util/null";
+
+type Entry<T> = {map: RecursiveMap<T>, value: ?T};
+type RecursiveMap<T> = Map<string, Entry<T>>;
+class BaseTrie<K, T> {
```

Comment:
I would like to see the semantics documented, please. Perhaps consider:

```js
class Trie<K, V> {
  /**
   * Create an empty trie backed by the given address module.
   */
  constructor(addressModule: AddressModule<K>): void;

  /**
   * Add key `k` to this trie with value `v`. Return `this`.
   */
  add(k: K, v: V): this;

  /**
   * Get the values in this trie along the path to `k`.
   *
   * More specifically, this method has the following observable
   * behavior. Let `inits` be the list of all prefixes of `k`, ordered
   * by length (shortest to longest). Observe that the length of `inits`
   * is `n + 1`, where `n` is the number of parts of `k`; `inits` begins
   * with the empty address and ends with `k` itself. Initialize the
   * result to an empty array. For each prefix `p` in `inits`, if `p`
   * was added to this trie with value `v`, then append `v` to
   * `result`. Return `result`.
   */
  get(k: K): V[];
}
```

Reply:
Done